### PR TITLE
Port to 1.21

### PR DESCRIPTION
**Pull request by ThexXTURBOXx**
In my tests, everything seems to work fine. Marking as a draft until Minecraft 1.21 is properly released and NeoForge has final versions available

### CODE REVIEW

File: common/src/main/java/me/shedaniel/clothconfig2/gui/GlobalizedClothConfigScreen.java [hunk 1]
```diff
@@ -243,55 +242,51 @@ public void render(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
...
+            buffer.addVertex(matrix, sliderPosition - 14, 0, 100.0F).setColor(0, 0, 0, shadeColor);
+            buffer.addVertex(matrix, sliderPosition - 14 - 4, 0, 100.0F).setColor(0, 0, 0, 0);
+            buffer.addVertex(matrix, sliderPosition - 14 - 4, height, 100.0F).setColor(0, 0, 0, 0);
+            buffer.addVertex(matrix, sliderPosition - 14, height, 100.0F).setColor(0, 0, 0, shadeColor);
             RenderSystem.disableBlend();
         }
         Rectangle slideArrowBounds = new Rectangle(sliderPosition - 14, 0, 14, height);
         {
-            MultiBufferSource.BufferSource immediate = MultiBufferSource.immediate(Tesselator.getInstance().getBuilder());
+            MultiBufferSource.BufferSource immediate = MultiBufferSource.immediate(Tesselator.getInstance().buffer);
```
Comment (ThexXTURBOXx) [inline common/src/main/java/me/shedaniel/clothconfig2/gui/GlobalizedClothConfigScreen.java]: Is there a better way? Needs access widener/transformer

**Comment by shedaniel** [comment]
Thanks for this PR! I will take over from here!

**Review by shedaniel — APPROVED** [review]

File: common/src/main/java/me/shedaniel/clothconfig2/gui/GlobalizedClothConfigScreen.java (hunk 1 above)
Comment (shedaniel) [reply common/src/main/java/me/shedaniel/clothconfig2/gui/GlobalizedClothConfigScreen.java]: Yep, I have changed this to not required the AW/AT

File: gradle.properties [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-mod_menu_version=10.0.0-beta.1
+fabric_loader_version=0.15.11
+fabric_api_version=0.100.1+1.21
+mod_menu_version=11.0.0-beta.1
 
 forge_version=49.0.3
-neoforge_version=20.5.0-beta
-neoforge_pr=
+neoforge_version=21.0.0-alpha.1.21-rc1.20240613.054156
+neoforge_pr=1076
```
Comment (MattSturgeon) [inline gradle.properties]: Release is now available: https://github.com/neoforged/NeoForge/commit/dc521863d377a8be232488e6a57dd1eb758928f6

Comment (shedaniel) [reply gradle.properties]: Thank you for notifying me!

**Pull request merged**